prometheus/src/main/scala/filodb/prometheus/parse/Parser.scala

```diff
@@ -387,7 +387,8 @@ trait Expression extends Aggregates with Selector with Numeric with Join {
 
   lazy val expression: PackratParser[Expression] =
     binaryExpression | subqueryExpression | aggregateExpression2 | aggregateExpression1 |
-      function | unaryExpression | vector | numericalExpression | simpleSeries | precedenceExpression
+      function | unaryExpression | vector | numericalExpression | simpleSeries | precedenceExpression |
+      "(" ~> expression <~ ")"
```

Comment:
What does this magic syntax mean?

Reply:
It is for queries like `(sum(some_metric))`

Reply:
I mean, the magic "~>" and "<~" symbols. What do they mean?

Reply:
According to parsing.combinator, `p ~> q succeeds if `p` succeeds and `q` succeeds on the input  left over by `p`. `q` gets executed only if p is true and it returns result of q. Similarly `p <~ q` returns the result of p on success

https://github.com/scala/scala-parser-combinators/blob/2e43707531264ef30d0a623c5189939ff54841ca/shared/src/main/scala/scala/util/parsing/combinator/Parsers.scala#L307

Reply:
👍